firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -329,50 +329,35 @@ public SnapshotVersion getLastRemoteSnapshotVersion() {
     return persistence.runTransaction(
         "Apply remote event",
         () -> {
+          SnapshotVersion remoteVersion = remoteEvent.getSnapshotVersion();
+          Map<Integer, TargetChange> targetChanges = remoteEvent.getTargetChanges();
           long sequenceNumber = persistence.getReferenceDelegate().getCurrentSequenceNumber();
-          Set<DocumentKey> authoritativeUpdates = new HashSet<>();
 
-          Map<Integer, TargetChange> targetChanges = remoteEvent.getTargetChanges();
           for (Map.Entry<Integer, TargetChange> entry : targetChanges.entrySet()) {
             Integer boxedTargetId = entry.getKey();
             int targetId = boxedTargetId;
             TargetChange change = entry.getValue();
 
-            // Do not ref/unref unassigned targetIds - it may lead to leaks.
-            QueryData queryData = targetIds.get(targetId);
-            if (queryData == null) {
-              continue;
-            }
-
-            // When a global snapshot contains updates (either add or modify) we can completely
-            // trust these updates as authoritative and blindly apply them to our cache (as a
-            // defensive measure to promote self-healing in the unfortunate case that our cache
-            // is ever somehow corrupted / out-of-sync).
-            //
-            // If the document is only updated while removing it from a target then watch isn't
-            // obligated to send the absolute latest version: it can send the first version that
-            // caused the document not to match.
-            for (DocumentKey key : change.getAddedDocuments()) {
-              authoritativeUpdates.add(key);
-            }
-            for (DocumentKey key : change.getModifiedDocuments()) {
-              authoritativeUpdates.add(key);
-            }
-
-            queryCache.removeMatchingKeys(change.getRemovedDocuments(), targetId);
-            queryCache.addMatchingKeys(change.getAddedDocuments(), targetId);
-
-            // Update the resume token if the change includes one. Don't clear any preexisting
-            // value.
-            ByteString resumeToken = change.getResumeToken();
-            if (!resumeToken.isEmpty()) {
-              QueryData oldQueryData = queryData;
-              queryData =
-                  queryData.copy(remoteEvent.getSnapshotVersion(), resumeToken, sequenceNumber);
-              targetIds.put(boxedTargetId, queryData);
-
-              if (shouldPersistQueryData(oldQueryData, queryData, change)) {
-                queryCache.updateQueryData(queryData);
+            QueryData oldQueryData = targetIds.get(targetId);
+            if (oldQueryData != null) {
```

Comment:
Optional: I personally liked the `continue` approach that was in place before (this function is long, and I feel the increased nestedness hurts readability).

Reply:
Sounds good. Reverted.